### PR TITLE
Nats streaming support

### PR DESCRIPTION
**Pull request by blinktag**

### CODE REVIEW

File: cli/nats-streaming.go [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+		StringVar(&opts.NatsStreaming.ClusterID)
+
+	cmd.Flag("client-id", "User specified client ID to connect with").
+		Required().
+		StringVar(&opts.NatsStreaming.ClientID)
+
+	cmd.Flag("channel", "NATS stream channel. Ex: \"orders\"").
+		StringVar(&opts.NatsStreaming.Channel)
+
+	cmd.Flag("tls-ca-file", "CA file (only needed if addr is tls://").ExistingFileVar(&opts.NatsStreaming.TLSCAFile)
```
Comment (dselans) [inline cli/nats-streaming.go:62]:
```suggestion
	cmd.Flag("tls-ca-file", "CA file (only needed if addr is tls://)").ExistingFileVar(&opts.NatsStreaming.TLSCAFile)
```

Comment (dselans) [reply cli/nats-streaming.go:62]: One character PR update! 💪

Comment (dselans) [reply cli/nats-streaming.go:62]: + add a newline

File: backends/nats-streaming/read.go [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+		}
+
+		n.printer.Print(str)
+
+		if !n.Options.ReadFollow {
+			doneCh <- true
+		}
+	}
+
+	sub, err := subConn.Subscribe(n.Options.NatsStreaming.Channel, subFunc, n.getReadOptions()...)
```
Comment (dselans) [inline backends/nats-streaming/read.go:102]: Is this a non-blocking call, so you hit the read from `<-doneCh`?

Comment (dselans) [reply backends/nats-streaming/read.go:102]: Seems like it - nice design!

**Review by dselans — APPROVED** [review]
Really solid!

File: cli/nats-streaming.go (hunk 1 above)
Comment (blinktag) [reply cli/nats-streaming.go:62]: Fixed!

File: backends/nats-streaming/read.go (hunk 2 above)
Comment (blinktag) [reply backends/nats-streaming/read.go:102]: Yup! Simplest way to handle this case I think

**Pull request merged**